ntbs-service/Pages/LegacyNotifications/Index.cshtml.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using ntbs_service.DataMigration;
+using ntbs_service.Models;
+using ntbs_service.Services;
+
+namespace ntbs_service.Pages.LegacyNotifications
+{
+    public class Index : PageModel
+    {
+        private readonly ILegacySearchService _legacySearchService;
+        private readonly INotificationImportService _notificationImportService;
+
+        [BindProperty] 
+        public NotificationBannerModel NotificationBanner { get; set; }
+        
+        [BindProperty(SupportsGet = true)]
+        public string LegacyNotificationId { get; set; }
+        
+        public string RequestId { get; set; }
+        public ImportResult LegacyImportResult { get; set; }
+
+        public Index(ILegacySearchService legacySearchService, INotificationImportService notificationImportService)
+        {
+            _legacySearchService = legacySearchService;
+            _notificationImportService = notificationImportService;
+        }
+        
+        public async Task<IActionResult> OnGetAsync()
+        {
+            return await GetPage();
+        }
+
+        public async Task<IActionResult> OnPostAsync()
+        {
+            RequestId = HttpContext.TraceIdentifier;
+            var idsList = new List<string> {LegacyNotificationId};
+            ImportResult importResult = (await _notificationImportService.ImportByLegacyIdsAsync(null, RequestId, idsList)).FirstOrDefault();
```

Comment:
Again, using `Single` seems correct here so we don't need the `null` check below?